init.lua

```diff
@@ -244,27 +244,35 @@ function boost_cart.cart:on_step(dtime)
 		-- Slow down or speed up..
 		local acc = dir.y * -1.8
 
-		local speed_mod = tonumber(minetest.get_meta(pos):get_string("cart_acceleration"))
-		if speed_mod and speed_mod ~= 0 then
-			if speed_mod > 0 then
-				for _,v in ipairs({"x","y","z"}) do
-					if math.abs(vel[v]) >= max_vel then
-						speed_mod = 0
-						break
+		local halt = minetest.get_meta(pos):get_string("cart_halt") == "1"
```

Comment:
Merge the meta string `cart_halt` into `cart_acceleration` because these are conflicting parameters: `cart_halt` automatically means you have no acceleration. Thus, it would make sense to check if `cart_acceleration` is set to `halt`.